src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -130,6 +129,17 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
                            .Select(ToPackageDownloadDependency));
                 }
 
+                if (targetFrameworkInfo is IVsTargetFrameworkInfo3 targetFrameworkInfo3 && cpvmEnabled)
+                {
+                    if (targetFrameworkInfo3.CentralPackageVersions != null)
+                    {
+                        tfi.CentralPackageVersions.AddRange(
+                           targetFrameworkInfo3.CentralPackageVersions
+                               .Cast<IVsReferenceItem>()
+                               .Select(ToCentralPackageVersion).ToDictionary(cpv => cpv.Name));
```

Comment:
`ToDictionary` will cause a nomination failure if there are multiple `PackageVersion` items with the same ID/name.

Reply:
Good catch - updated.